Frameworks/UIKit/UILabel.mm

```diff
@@ -44,61 +44,106 @@ @implementation UILabel {
     UILineBreakMode _lineBreakMode;
     BOOL _adjustFontSize;
     float _minimumFontSize;
-    float _originalFontSize;
     int _numberOfLines;
     BOOL _isDisabled;
     BOOL _isHighlighted;
     UIBaselineAdjustment _baselineAdjustment;
     StrongId<WXCTextBlock> _textBlock;
 }
 
-- (void)adjustFontSizeToFit {
-    if (_numberOfLines == 0) {
-        [self adjustTextLayerSize];
-        return;
-    }
-
-    float curFontSize = _originalFontSize;
-    _font = [_font fontWithSize:curFontSize];
-
-    CGRect rect;
-    rect = [self bounds];
-    if (rect.size.width == 0.0f || rect.size.height == 0.0f) {
-        return;
-    }
+// Helper searches max font size for a string that can fit into given rect using given font/linkbreak config .
+// if no fit is found (e.g., to satisfy minimum font size requirement, nil is returned.
+// Otherwise, it returns the maximum font size for that can fit
+- (UIFont*)_findMaxFontSizeToFit:(CGRect)rect
```

Comment:
i c. let me do that in UILabel work in this comming iteration. I am expecting a lot more tests will be written for UILabel. and this function should be move to somewhere that can be shared by multiple controls.

---
In reply to: [98065011](https://github.com/Microsoft/WinObjC/pull/1768#discussion_r98065011) [](ancestors = 98065011,98056283,98052549)